common/src/main/scala/com/theguardian/multimedia/archivehunter/common/ProxyTranscodeFramework/ProxyGenerators.scala

```diff
@@ -187,14 +193,69 @@ class ProxyGenerators @Inject() (config:ArchiveHunterConfiguration,
           jobModelDAO.deleteJob(jobDesc.jobId)  //ignore the result, this is non-essential but there to prevent the jobs list getting clogged up
           Future(Failure(NothingFoundError("media", "Nothing found to proxy")))
         case Some(uriString) =>
-          val rq = RequestModel(requestType,uriString,targetProxyBucket,jobUuid.toString,None,None,proxyType)
+          val rq = RequestModel(requestType,uriString,targetProxyBucket,jobDesc.jobId,None,None,proxyType)
           sendRequest(rq, target.region)
       }
     }).recoverWith({
       case err:Throwable=>
         logger.error("Could not start proxy job: ", err)
         Future(Failure(ExternalSystemError("archivehunter", err.toString)))
     })
+
+  /**
+    * tries to kick off a "stuck in pending" job again by resubmitting
+    * @param jobUuid UUId of the job to redo
+    * @param proxyType the ProxyType is not stored, so we have to request it again here. if None, the default for the type of media
+    *                  referenced by the job will be used
+    * @return a Future with either an error string or the job ID as a string.
+    */
+  def rerunProxyJob(jobUuid:UUID) = {
+    val entryFuture = jobModelDAO.jobForId(jobUuid.toString).flatMap({
+      case None=>Future(Left(s"No job found for ${jobUuid.toString}"))
+      case Some(Left(err))=>Future(Left(err.toString))
+      case Some(Right(jobModel))=>
+        indexer.getById(jobModel.sourceId).map(result=>Right((jobModel,result)))
+    })
+
+    entryFuture.flatMap({
+      case Left(err)=>Future(Left(err))
+      case Right(params)=>
+        val jobModel = params._1
+        val entry = params._2
+
+        val requestType = jobModel.jobType match {
+          case "thumbnail"=>Right(RequestType.THUMBNAIL)
```

Comment:
requestType should be in the TranscodeInfo section of JobModel then we won't need this here and can simplify